src/index.ts

```diff
@@ -118,6 +124,17 @@ const builder = <a, b>(
     }
     return entry.handler(value, entry.select(value));
   },
+
+  /**
+   * ### exhaustiveMatch
+   * Entry point to create pattern matching code branches. It returns an
+   * empty Match case.
+   */
+  exhaustive: (): ExhaustiveMatch<DistributeUnions<a>, b> =>
```

Comment:
This should only be allowed right after calling `match(...)`. It doesn't make sense if you already appended a `.with()`. Maybe I we should expose this as another top level `exhaustiveMatch(...)` function, as @AndrewMorsillo suggested? that might make more sense

Reply:
+1, I like the idea.

Maybe just an `exhaustive()`  top level function ? Or `ematch()` ? (`exhaustiveMatch` sounds a bit cumbersome to me for a name that will be heavily used, but I guess its a matter of opinion 😄)